library/src/lib/time/time.component.ts

```diff
@@ -256,97 +277,85 @@ export class TimeComponent implements OnChanges, ControlValueAccessor {
+    /** @hidden
+     * Handles period model change. depending on current hour and new period changes hours by +/- 12
+     * */
+    periodModelChange(): void {
+        if (this.time && !this.time.hour) {
+            this.time.hour = 0;
+        }
+        if (!this.period ||
+            (!this.isPm(this.period) && !this.isAm(this.period))
+        ) {
             this.periodInvalid = true;
         } else if (this.time.hour < 24 && this.time.hour >= 0) {
-            if (this.oldPeriod === 'am' && this.period === 'pm') {
+            this.periodInvalid = false;
+            if (this.isPm(this.period) && this.time.hour < 12) {
                 this.time.hour = this.time.hour + 12;
-            } else if (this.oldPeriod === 'pm' && this.period === 'am') {
-                if (this.time.hour === null) {
-                    this.time.hour = 0;
-                } else {
-                    this.time.hour = this.time.hour - 12;
-                }
+            } else if (this.time.hour >= 12 && this.isAm(this.period)) {
+                this.time.hour = this.time.hour - 12;
             }
-            this.periodInvalid = false;
             this.onChange(this.time);
         }
-        this.setDisplayedHour();
     }
 
-    /** @hidden */
-    registerOnChange(fn: (time: TimeObject) => void): void {
-        this.onChange = fn;
-    }
-
-    /** @hidden */
-    registerOnTouched(fn: any): void {
-        this.onTouched = fn;
-    }
-
-    /** @hidden */
-    setDisabledState(isDisabled: boolean): void {
-        this.disabled = isDisabled;
-    }
-
-    /** @hidden */
-    writeValue(time: TimeObject): void {
-        if (!time) {
-            return;
-        }
-        this.time = time;
-        this.setDisplayedHour();
-    }
-
-    /** @hidden */
-    lastButtonKeydown(event) {
+    /** @hidden
+     * Handles last button keyboard events
+     * */
+    lastButtonKeydown(event: KeyboardEvent): void {
         if (event.code === 'Tab' && !event.shiftKey) {
             event.preventDefault();
             this.focusArrowLeft.emit();
         }
     }
 
-    constructor(public timeI18nLabels: TimeI18nLabels) {}
+    /**
+     * @hidden
+     * Defines if period is PM, Considers the fact that period should be case sensitive
+     * */
+    private isPm(period: string): boolean {
+        const pmMeridian = this.timeI18n.meridianCaseSensitive ? this.timeI18n.meridianPm : this.timeI18n.meridianPm.toLowerCase();
+        period = this.timeI18n.meridianCaseSensitive ? period : period.toLowerCase();
+        return period === pmMeridian;
+    }
+
+    /**
+     * @hidden
+     * Defines if period is AM, Considers the fact that period should be case sensitive
+     * */
+    private isAm(period: string): boolean {
+        const amMeridian = this.timeI18n.meridianCaseSensitive ? this.timeI18n.meridianAm : this.timeI18n.meridianAm.toLowerCase();
+        period = this.timeI18n.meridianCaseSensitive ? period : period.toLowerCase();
```

Comment:
Anywhere you use `toLowerCase()` instead use `toLocaleLowerCase()`.

Actually, it might even be better to do the comparison in `toLocaleUpperCase()`